src/installer/tests/HostActivation.Tests/NativeHosting/FunctionPointerLoading.cs

```diff
@@ -0,0 +1,219 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build.Framework;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public partial class FunctionPointerLoading : IClassFixture<ComponentActivation.SharedTestState>
+    {
+        private const string FunctionPointerActivationArg = "get_function_pointer";
+        private const string FunctionPointerFromAppActivationArg = "app_get_function_pointer";
+
+        private readonly ComponentActivation.SharedTestState sharedState;
+
+        public FunctionPointerLoading(ComponentActivation.SharedTestState sharedTestState)
+        {
+            sharedState = sharedTestState;
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        public void CallDelegate(bool validType, bool validMethod)
```

Comment:
Please be explicit which context - this one means "app" implicitly, in the product you use "component" implicitly... I personally don't mind long descriptive names, so `CallDelegateOnApplicationContext` for example...